source/array-includes.ts

```diff
@@ -0,0 +1,24 @@
+/**
+A better version `Array.prototype.includes` which allows other types to be included and also acts as a type guard when the provided array is of a specific type.
+
+This is needed because `Array.prototype.includes` doesn't allow for other types to be included.
```

Comment:
Can you link to a TS issue about this?

Reply:
The proposal was rejected in this issue: https://github.com/microsoft/TypeScript/issues/26255#issuecomment-748211891